pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/AggregateOperator.java

```diff
@@ -69,18 +74,22 @@ public class AggregateOperator extends BaseOperator<TransferableBlock> {
   private boolean _readyToConstruct;
   private boolean _hasReturnedAggregateBlock;
 
+  // TODO: Move to OperatorContext class.
+  private OperatorStats _operatorStats;
+
   // TODO: refactor Pinot Reducer code to support the intermediate stage agg operator.
   // aggCalls has to be a list of FunctionCall and cannot be null
   // groupSet has to be a list of InputRef and cannot be null
   // TODO: Add these two checks when we confirm we can handle error in upstream ctor call.
   public AggregateOperator(Operator<TransferableBlock> inputOperator, DataSchema dataSchema,
-      List<RexExpression> aggCalls, List<RexExpression> groupSet) {
-    this(inputOperator, dataSchema, aggCalls, groupSet, AggregateOperator.Accumulator.MERGERS);
+      List<RexExpression> aggCalls, List<RexExpression> groupSet, PlanRequestContext context) {
```

Comment:
can we create the OperatorStats at the PhysicalPlanVisitor or passing in the requestID and stageId directly? is there any other info needed from the context?

Reply:
Done